lib/OpenQA/WebAPI/Plugin/ObsRsync/Runner.pm

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2019 SUSE Linux GmbH
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package OpenQA::WebAPI::Plugin::ObsRsync::Runner;
+use strict;
+use warnings;
+use IPC::System::Simple qw(system $EXITVAL);
+
+sub newRunner {
+    my $app = shift;
+    $app->minion->add_task(
+        obs_rsync => sub {
+            my ($job, $args) = @_;
+            eval { system([0], "bash", @$args); 1 };
+            $app->minion->unlock('obs_rsync_lock');
+            $job->finish($EXITVAL);
+        });
+}
+
+my $lock_timeout = 3600;
+
+sub Run {
```

Comment:
It should be `run` (lower case). Or just inline that function. Having a module for just 2 helper functions seems a bit of an overkill. Especially because if you use `$app->gru->enqueue_and_keep_track(...)` as suggested in the other comment this code should become much simpler.